smac/runhistory/runhistory.py

```diff
@@ -328,9 +333,20 @@ def save_json(self, fn: str="runhistory.json", save_external: bool=False):
                    for id_, conf in self.ids_config.items()
                    if id_ in config_ids_to_serialize}
 
-        with open(fn, "w") as fp:
-            json.dump({"data": data,
-                       "configs": configs}, fp, cls=EnumEncoder)
+        # add lock object for `fn` if it does not exist
+        try:
+            if fn not in RunHistory.rw_locks:
+                RunHistory.rw_locks[fn] = filelock.FileLock(fn + ".lock")
+        except Exception as e:
+            logging.error(e)
```

Comment:
I don't think it is a good idea to catch an exception here if it is not handled. The program should rather crash if there is an issue instead of continuing in an undefined state.